exercises/concept/poetry-club-door-policy/door-policy.spec.js

```diff
@@ -7,239 +9,117 @@ import {
   backDoorPassword,
 } from './door-policy';
 
-class Poem {
-  constructor(poem) {
-    const [lines, author] = poem.split('\n\n');
-    this.lines = lines.split('\n');
-    this.author = author;
-  }
-
-  get length() {
-    return this.lines.length;
-  }
-
-  // purposefully using array methods as to not give away string solution
-  get acrostic() {
-    return this.lines.map((line) => [...line].shift()).join('');
-  }
-
-  get telestich() {
-    return this.lines.map((line) => [...line.trim()].pop()).join('');
-  }
-
-  *[Symbol.iterator]() {
-    yield* this.lines[Symbol.iterator]();
-  }
-}
-
-class FrontDoorGuard {
-  /**
-   * @param {Poem} poem
-   */
-  constructor(poem) {
-    this.poem = poem;
-  }
-
-  /**
-   * Recites a poem at the door, asking the patron for a response
-   *
-   * @returns {string} key
-   */
-  recite() {
-    const patron = frontDoorPatron;
-    const key = [];
-
-    for (const line of this.poem) {
-      key.push(patron(line));
-    }
-
-    return key.join('');
-  }
-
-  assert() {
-    const password = frontDoorPassword(this.recite());
-
-    expect(password).toBe(stringify(generateCapitalized(this.poem.acrostic)));
-  }
-}
-
-class BackDoorGuard {
-  constructor(poem) {
-    this.poem = poem;
-  }
-
-  /**
-   * Recites a poem at the door, asking the patron for a response
-   *
-   * @returns {string} key
-   */
-  recite() {
-    const patron = backDoorPatron;
-    const key = [];
-
-    for (const line of this.poem) {
-      key.push(patron(line));
-    }
-
-    return key.join('');
-  }
-
-  assert() {
-    const password = backDoorPassword(this.recite());
-
-    expect(password).toBe(
-      stringify(generateCapitalized(this.poem.telestich)).concat(', please')
-    );
-  }
-}
+const recite = (poem, patron) => {
+  return poem.map((line) => patron(line));
+};
 
-const SHIRE_HORSE = new Poem(
-  `
+//Shire Horse, by Michael Lockwood
+const SHIRE_HORSE = `
 Stands so high
 Huge hooves too
 Impatiently waits for
 Reins and harness
 Eager to leave
+`
+  .trim()
+  .split('\n');
 
-Michael Lockwood
-`.trim()
-);
-
-const SHIRE_HORSE_WITH_SPACES = new Poem(
-  `
-Stands so high  
-Huge hooves too      
-Impatiently waits for 
-Reins and harness  
+//Shire Horse, by Michael Lockwood
+const SHIRE_HORSE_WITH_SPACES = `
+Stands so high
```

Comment:
Since the variable name includes "spaces" and this test case should help the student figure out they need to trim, I would suggest adding some obvious spaces like shown below (instead of just new line whitespaces).
```suggestion
     Stands so high
```

Reply:
Restructured the poems to make them simpler and whitespace more obvious.